### PR TITLE
Port api/tools/fetch to FastAPI, add schema, and use `__FETCH_DATA__` in most tests

### PR DESCRIPTION
**Pull request by mvdbeek**
I think it's important that we target the newer `api/tools/fetch` in our tests. This has revealed a small issue that I fixed in here as well (4212c7590dad2ff4940661df5ca7b67318b8bd3a).
The new schema should be very helpful for anyone trying to target uploads via `api/tools/fetch`.

Screenshot 2022-04-25 at 12 37 34

The form data description doesn't properly render the nested `targets` model, but in practice you should upload files via TUS and post the result with `application/json` as the content type, so I'm leaving that as is for now.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by jmchilton — APPROVED** [review]

File: lib/galaxy/schema/fetch_data.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+class FetchBaseModel(BaseModel):
+    class Config:
+        allow_population_by_field_name = True
+
+
+class ElementsFromType(str, Enum):
+    ARCHIVE = "archive"
+    BAGIT = "bagit"
+    BAGIT_ARCHIVE = "bagit_archive"
+    DIRECTORY = "directory"
```
Comment (davelopez) [inline lib/galaxy/schema/fetch_data.py:34]: Maybe declare those enums in lowercase for consistency

File: lib/galaxy/schema/fetch_data.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+class BaseFetchDataTarget(FetchBaseModel):
+    auto_decompress: bool = AutoDecompressField
+
+
+class ItemsFromSrc(str, Enum):
+    url = "url"
+    files = "files"
+    path = "path"
+    ftp_import = "ftp_import"
+    server_dir = "server_id"
```
Comment (davelopez) [inline lib/galaxy/schema/fetch_data.py:49]: I think this is a typo
```suggestion
    server_dir = "server_dir"
```

File: lib/galaxy/webapps/galaxy/services/tools.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+        toolbox_search: ToolBoxSearch,
+        security: IdEncodingHelper,
+        history_manager: HistoryManager,
+    ):
+        super().__init__(security)
+        self.config = config
+        self.toolbox_search = toolbox_search
+        self.history_manager = history_manager
+
+    def _create_fetch(
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/services/tools.py:54]: The service methods that are going to be used by the API controller (somewhat *public*) should probably be declared without `_`, not a big deal anyway.

**Review by davelopez — APPROVED** [review]
Impressive! :+1: 

Just some minor comments below.

**Comment by davelopez** [comment]
Oops! took me too long to review sorry! :sweat_smile:

**Comment by mvdbeek** [comment]
oops, I should have waited just a bit longer. I'll followup

**Pull request merged**